Makefile

```diff
@@ -18,10 +20,20 @@ get-deps-pip:
 
 .PHONY: get-deps-go
 get-deps-go:
+	$(MAKE) go-version-check
 	$(MAKE) get-deps-master
 	$(MAKE) get-deps-agent
 	$(MAKE) get-deps-proto
 
+# Go versions may look like goM, goM.N, or goM.N.P. Only 1.18.* is supported.
+supported_go_minor_version = go1.18
+system_go_version := $(shell go version | sed 's/.*\(go[[:digit:]][[:digit:]0-9.]*\).*/\1/')
+.PHONY: go-version-check
+go-version-check:
+	@: $(if $(findstring $(supported_go_minor_version), $(system_go_version)), \
+				, \
+				$(error go version $(system_go_version) not supported. Must use $(supported_go_minor_version).x))
```

Comment:
This certainly gets the job done, though I must say I was envisioning something a bit more straightforward like `@go version | grep -q 'go1\.18' || (echo 'need go 1.18!' && exit 1)`. Either way works.

Reply:
I think I prefer this maybe-overengineering approach to the maybe-underengineering approach. The sed regex is awful, but so long as it's trustable, it just feels more "right" to treat make as a language and perform as much logic in that as possible.

But I'm very much unfamiliar with make idioms. If you're confident that the simpler approach better reflects how people expect make to be used, I'll totally make this change.

Reply:
Mmm. Speaking of that awful regex, I'm noticing I kind of chose something in-between. Regexes in make aren't great, but I could parse out the go version there from the string returned by `go version`. Worth it?